extensions/csv/src/main/java/io/deephaven/csv/tokenization/Tokenizer.java

```diff
@@ -0,0 +1,530 @@
+            final int savedBegin = bs.begin();
+            if (bs.begin() == bs.end()) {
+                return false;
+            }
+            final char front = (char) bs.front();
+            boolean negative = false;
+            if (front == '+') {
+                bs.setBegin(bs.begin() + 1);
+            } else if (front == '-') {
+                negative = true;
+                bs.setBegin(bs.begin() + 1);
+            }
+            if (!tryParseWholeNumber(bs, 1, 999, negative, result)) {
+                bs.setBegin(savedBegin);
+                return false;
+            }
+            // Successful iff completely consumed.
+            if (bs.begin() != bs.end()) {
+                bs.setBegin(savedBegin);
+                return false;
+            }
+            return true;
+        }
+
+        // Allowable formats in DH style (tz stands for some supported timezone like NY)
+        // 2021-11-07T09:00:00 tz
+        // 2021-11-07T09:00:00.1 tz
+        // 2021-11-07T09:00:00.12 tz
+        // ...
+        // 2021-11-07T09:00:00.123456789 tz
+        //
+        // hyphens and colons are optional (all in or all out)
+        // The 'T' can also be a space.
+        //
+        // Instead of " tz" we also admit Z (no space) meaning UTC
+        //
+        // Allowable formats in UTC offset style (can be + or -)
+        // The offset can be hh or hh:mm or hhmm
+        // 2021-11-07T09:00:00+01
+        // 2021-11-07T09:00:00.1-02:30
+        // 2021-11-07T09:00:00.12+0300
+        // ...
+        // 2021-11-07T09:00:00.123456789+01:30
+        //
+        // Putting it all together: First we have the required fields
+        // yyyy
+        // - (optional, but if absent then no later hyphens or colons)
+        // mm
+        // - (optional, see above)
+        // dd
+        // T
+        // hh
+        // : (optional, see above)
+        // MM
+        // : (optional, see above)
+        // SS
+        // . (optional fraction, must be followed by 1-9 decimal digits) (can also be comma apparently)
+        // Z or space or + or - :
+        // Z means UTC
+        // space means there is a Deephaven time zone indicator following, which must be one of the
+        // enumeration values in the class DBTimeZone
+        // + or - means an offset follows, which itself is
+        // hh
+        // : (optional)
+        // mm (optional)
+        private static boolean tryParseDBDateTime(
+                ByteSlice bs, TimeZoneParser tzp,
+                long[] temp0, long[] temp1, long[] temp2, DateTimeZone[] tempZone,
+                long[] result) {
+            final int savedBegin = bs.begin();
+            if (!tryParseYyyymmdd(bs, temp0, temp1, temp2)) {
+                return false;
+            }
+            final int year = (int) temp0[0];
+            final int month = (int) temp1[0];
+            final int day = (int) temp2[0];
+
+            // Require 'T' (or, per RFC 3339, allow ' ')
```

Comment:
Great that we support 3339, I had some issues with reading data produced from python pandas/arrow due to not supporting `' '` in the old code.

Reply:
👍 👍